clientv1.go

```diff
@@ -135,20 +119,26 @@ func (c *ClientV1) Tokens() *TokensServiceV1 {
 	return &TokensServiceV1{client: c}
 }
 
-// TokenSource returns a valid access token to send requests authenticated with
-// a SAMS access token. Internally, the token returned is reused. So that new
-// tokens are only created when needed. (Provided this `Client` is long-lived.)
-//
-// To send outbound requests to other SAMS clients, you would use:
-//
-//	```go
-//	httpClient := oauth2.NewClient(ctx, samsClient.TokenSource())
-//	```
-func (c *ClientV1) TokenSource() oauth2.TokenSource {
-	return c.clientCredentialsConfig.TokenSource(context.Background())
-}
-
 var (
 	ErrNotFound       = errors.New("not found")
 	ErrRecordMismatch = errors.New("record mismatch")
 )
+
+// ClientCredentialsTokenSource returns a TokenSource that generates an access
+// token using the client credentials flow. Internally, the token returned is
+// reused. So that new tokens are only created when needed. (Provided this
+// `Client` is long-lived.)
+//
+// The `requestScopes` is a list of scopes to be requested for this token
+// source. Scopes should be defined using the available scopes package. All
+// requested scopes must be allowed by the registered client - see:
+// https://sourcegraph.notion.site/6cc4a1bd9cb247eea9674dbf9d5ce8c3
+func ClientCredentialsTokenSource(apiURL, clientID, clientSecret string, requestScopes []scopes.Scope) oauth2.TokenSource {
```

Comment:
Since this can be used for all APIs, should we put this in `sams.go` instead?

Reply:
Syntactically yes but semantically a client-credentials-issued access tokens can really only be used for `ClientV1`, right?

Reply:
I'm not sure, NGL I it would _make sense_ to allow client credentials with very escalated access to get user account details via accounts APIs as well (internal automation etc) but maybe I'm very off the mark here 😁

Reply:
>allow client credentials with very escalated access to get user account details via accounts APIs

Is that the right way to think about this? 😂 client-credentials-issued access tokens are not tied to any particular users (and it can't issue per-user access tokens other wise it breaks the responsibility boundary of these two sets of APIs), thus it will/should never pass authN check on Accounts API (which we can make such thing to work but I'd suggest against that, and instead, add needed API endpoints in Clients APIs).

Reply:
> thus it will/should never pass authN check on Accounts API (which we can make such thing to work but I'd suggest against that, and instead, add needed API endpoints in Clients APIs).

Separate APIs works as well :) But it could be handy to provide escalated access to various APIs as well

Also FYI, problem with the constructor that I missed during review: https://github.com/sourcegraph/sourcegraph-accounts-sdk-go/pull/17

Reply:
> But it could be handy to provide escalated access to various APIs as well

Wouldn't disagree it could offer connivence on the client side, but I think mixing principal type (i.e. am I a user or client? or both?) dramatically complicates the server-side (assumption and thus) implementation, and given security sensitive characteristics of SAMS, I personally feel the risk is not worth it.